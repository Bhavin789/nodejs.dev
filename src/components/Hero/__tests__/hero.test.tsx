/* eslint-disable @typescript-eslint/explicit-function-return-type */
import React from 'react';
import renderer from 'react-test-renderer';

import Hero from '..';

const mockReleaseData = [
  {
    version: 'mock-version-1',
    date: 'mock-date',
    files: ['aix-ppc64', 'headers', 'linux-arm64'],
    npm: 'mock-npm',
    v8: 'mock-v8',
    uv: 'mock-uv',
    zlib: 'mock-zlib',
    openssl: 'mock-openssl',
    modules: 'mock-module',
    lts: true,
    security: false,
  },
  {
    version: 'mmock-version-2',
    date: 'mock-date',
    files: ['aix-ppc64', 'headers', 'linux-arm64', 'linux-armv7l'],
    npm: 'mock-npm',
    v8: 'mock-v8',
    uv: 'mock-uv',
    zlib: 'mock-zlib',
    openssl: '1.1.1g',
    modules: '83',
    lts: true,
    security: false,
  },
];

jest.mock('../../../hooks/useReleaseHistory', () => ({
  useReleaseHistory: () => mockReleaseData,
}));

describe('Hero component', () => {
  it('renders correctly', () => {
    const title = 'Introduction to Node.js';
    const subTitle = 'Mock SubTitle';
    const tree = renderer
      .create(<Hero title={title} subTitle={subTitle} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
