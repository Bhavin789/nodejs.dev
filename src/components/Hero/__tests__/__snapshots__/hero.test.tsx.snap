// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component renders correctly 1`] = `
<div
  className="home-page-hero"
>
  <h1>
    Introduction to Node.js
  </h1>
  <h2
    className="sub-title t-subheading"
  >
    Mock SubTitle
  </h2>
  <div
    className="btn-ctas"
  >
    <div
      className="download-lts-container"
    >
      <a
        className="circular-container"
        href="https://nodejs.org/dist/mmock-version-2/node-mmock-version-2.tar.gz"
      >
        Download Node (LTS)
      </a>
      <p
        className="t-caption"
      >
        Version mock-version-2 - 
        <a
          href="https://nodejs.org/dist/mock-version-1/node-mock-version-1.tar.gz"
        >
          Get Current
        </a>
      </p>
    </div>
    <a
      className="circular-container inverse"
      href="/learn"
    >
      Learn Node
    </a>
  </div>
</div>
`;
